test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,213 @@
+                                             std::ranges::range_value_t<rng_t>>;
+    using fm_index_seqan3_t = decltype(seqan3::fm_index{std::declval<rng_t>()});
+    using bi_fm_index_seqan3_t = decltype(seqan3::bi_fm_index{std::declval<rng_t>()});
+
+    inner_rng_t make_single_seqan3(size_t seed = 0)
+    {
+        // The characters with rank 254 and 255 are reserved for the indices.
+        if constexpr (seqan3::alphabet_size<alphabet_t> > 254)
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed)
+                   | seqan3::views::persist
+                   | seqan3::views::to_rank
+                   | std::views::transform([] (uint8_t const r)
+                   {
+                       return (r > 253 ? 253 : r);
+                   })
+                   | seqan3::views::rank_to<alphabet_t>
+                   | seqan3::views::to<inner_rng_t>;
+        }
+        else
+        {
+            return seqan3::test::generate_sequence<alphabet_t>(length, 0, seed) | seqan3::views::to<inner_rng_t>;
+        }
+    }
+
+    rng_t make_collection_seqan3()
+    {
+        rng_t result;
+        for (size_t i = 0; i < collection_size; ++i)
+            result.emplace_back(make_single_seqan3(i));
+
+        return result;
+    }
+
+    rng_t make_sequence_seqan3()
+    {
+        if constexpr(seqan3::dimension_v<rng_t> == 1)
+            return make_single_seqan3();
+        else
+            return make_collection_seqan3();
+    }
+
+#if SEQAN3_HAS_SEQAN2
+    // Map the SeqAn3 alphabet to its SeqAn2 equivalent.
+    using seqan2_alphabet_t = std::conditional_t<std::same_as<alphabet_t, seqan3::dna4>, seqan::Dna,
+                                std::conditional_t<std::same_as<alphabet_t, seqan3::aa27>, seqan::AminoAcid, char>>;
+    // SeqAn2: `seqan::String` and `seqan::StringSet`, SeqAn3: `std::vector` and `std::vector<std::vector>`.
+    using seqan2_rng_t = std::conditional_t<seqan3::dimension_v<rng_t> == 1,
+                                            seqan::String<seqan2_alphabet_t>,
+                                            seqan::StringSet<seqan::String<seqan2_alphabet_t>>>;
+    using index_cfg = seqan::FastFMIndexConfig<void, uint32_t>;
+    using fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::FMIndex<void, index_cfg>>;
+    using bi_fm_index_seqan2_t = seqan::Index<seqan2_rng_t, seqan::BidirectionalIndex<seqan::FMIndex<void, index_cfg>>>;
+
+    template <typename to_t, typename from_t, template <typename> typename range_t>
+        requires (seqan3::alphabet<from_t> && std::ranges::forward_range<range_t<from_t>>)
+    seqan::String<to_t> convert_single_range(range_t<from_t> const & original_range)
+    {
+        seqan::String<to_t> result_range;
+        seqan::reserve(result_range, std::ranges::size(original_range));
+
+        for (auto const & element : original_range)
+            appendValue(result_range, seqan3::to_char(element));
+
+        return result_range;
+    }
```

Comment:
It seems like it possible to replace this whole function by 
```cpp
std::transform(begin(original_range), end(original_range), begin(result_range), seqan3::to_char)
```

Reply:
I could do
```
std::transform(std::ranges::begin(original_range), std::ranges::end(original_range), seqan::begin(result_range), seqan3::to_char);
```
but this is also two lines, and maybe harder to understand

Reply:
Yes, but I don't mean to put this code inside the function, but rather you could avoid defining the whole template function (which spans 12 lines) by using `transform` wherever you would otherwise call `convert_single_range`.
You want to make a to_char conversion and in my opinion `transform` should be intuitive enough. 

But I also notice that then you would have to change the calling code, because `transform` has different return semantics etc. so maybe it's more difficult than I think...